pkg/crdupdater/clusterrole.go

```diff
@@ -85,31 +76,36 @@ func (c *ClusterRole) buildNewAndOldObject() error {
 }
 
 // add adds the cluster role
-func (c *ClusterRole) add() error {
+func (c *ClusterRole) add(isPatched bool) (bool, error) {
 	isAdded := false
 	for _, clusterRole := range c.clusterRoles {
 		if _, ok := c.oldClusterRoles[clusterRole.GetName()]; !ok {
 			c.deployer.Deployer.AddClusterRole(clusterRole)
 			isAdded = true
 		}
 	}
-	if isAdded {
+	if isAdded && !c.config.dryRun {
 		err := c.deployer.Deployer.Run()
 		if err != nil {
-			return errors.Annotatef(err, "unable to deploy cluster role in %s", c.namespace)
+			return false, errors.Annotatef(err, "unable to deploy cluster role in %s", c.config.namespace)
 		}
 	}
-	return nil
+	return false, nil
```

Comment:
isPatched is required for patching the rc and deployment. in case of any change in cluster role, clusterrolebinding, service, the patch is not required. that's why for those it returns false. In case of configmap or secret changed, it will return true and then rc and deployment patch will be triggered.

Reply:
In this case you are always returning false though.  Shouldn't there be a case where this returns true?  What does the return value signify?